ext/http/01_http.js

```diff
@@ -290,13 +292,40 @@
               }
               await reader.cancel(error);
               throw error;
+            } finally {
+              readableStreamClose(respBody); // Release JS lock.
+            }
+          } else {
+            const reader = respBody.getReader();
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) break;
+              if (!ObjectPrototypeIsPrototypeOf(Uint8ArrayPrototype, value)) {
+                await reader.cancel(new TypeError("Value not a Uint8Array"));
+                break;
+              }
+              try {
+                await core.opAsync("op_http_write", streamRid, value);
+              } catch (error) {
+                const connError = httpConn[connErrorSymbol];
+                if (
+                  ObjectPrototypeIsPrototypeOf(BadResourcePrototype, error) &&
+                  connError != null
+                ) {
+                  // deno-lint-ignore no-ex-assign
+                  error = new connError.constructor(connError.message);
+                }
+                await reader.cancel(error);
+                throw error;
+              }
+            }
+
+            try {
+              await core.opAsync("op_http_shutdown", streamRid);
+            } catch (error) {
+              await reader.cancel(error);
```

Comment:
Does this throw if called after `readableStreamClose(respBody)`?

Reply:
This is unreachable since `op_http_write_resource` performs the shutdown itself.

Reply:
Oh right. 👍